src/stat/stat.rs

```diff
@@ -471,13 +471,19 @@ impl Stater {
         let default_dev_tokens = Stater::generate_tokens(&Stater::default_fmt(showfs, terse, true), use_printf)
             .unwrap();
 
-        let reader = BufReader::new(File::open(MOUNT_INFO).expect("Failed to read /etc/mtab"));
-        let mut mount_list = reader.lines()
-                                   .filter_map(|s| s.ok())
-                                   .filter_map(|line| line.split_whitespace().nth(1).map(|s| s.to_owned()))
-                                   .collect::<Vec<String>>();
-        // Reverse sort. The longer comes first.
-        mount_list.sort_by(|a, b| b.cmp(a));
+        let mount_list = if showfs {
+            // mount points aren't displayed when showing filesystem information
+            Vec::new()
```

Comment:
It may make more sense to change this to `None` (and thus make `mount_list` an `Option<Vec<String>>`) to avoid an unnecessary allocation.

Reply:
Good idea.  I should've thought of that. :)